plugin/src/main/java/com/iridium/iridiumskyblock/upgrades/OresUpgrade.java

```diff
@@ -1,18 +1,70 @@
 package com.iridium.iridiumskyblock.upgrades;
 
 import com.cryptomorin.xseries.XMaterial;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import lombok.NoArgsConstructor;
+import org.bukkit.Material;
 
 import java.util.Map;
 
 @NoArgsConstructor
 public class OresUpgrade extends UpgradeData {
-    public Map<XMaterial, Integer> ores;
-    public Map<XMaterial, Integer> netherOres;
+    @JsonSerialize
+    private Map<XMaterial, Integer> ores;
+    @JsonSerialize
+    private Map<XMaterial, Integer> netherOres;
+    @JsonIgnore
+    private Material[] oresArray;
+    @JsonIgnore
+    private Material[] netherOresArray;
+    @JsonIgnore
+    private boolean initialized = false;
 
     public OresUpgrade(int money, int crystals, Map<XMaterial, Integer> ores, Map<XMaterial, Integer> netherOres) {
         super(money, crystals);
         this.ores = ores;
         this.netherOres = netherOres;
     }
+
+    @JsonIgnore
+    public Material[] getOres() {
+        if (!initialized) initialize();
+        return oresArray;
+    }
+
+    @JsonIgnore
+    public Material[] getNetherOres() {
+        if (!initialized) initialize();
+        return netherOresArray;
+    }
+
+    @JsonIgnore
+    private void initialize() {
+        int arraySize = 0;
```

Comment:
Maybe choose a more descriptive variable name here. Sure, it's the size of an array, but that doesn't tell you anything about what it's used for